core/app/models/spree/promotion_handler/cart.rb

```diff
@@ -29,22 +29,26 @@ def activate
       end
 
       private
-        def promotions
-          promo_table = Promotion.arel_table
-          join_table = Arel::Table.new(:spree_orders_promotions)
 
-          join_condition = promo_table.join(join_table, Arel::Nodes::OuterJoin).on(
-            promo_table[:id].eq(join_table[:promotion_id])
-          ).join_sources
+      def promotions
```

Comment:
Was too high before also. And the new code actually scores lower than the old. So its an improvement and IMO not a merge blocker.